src/com/trilead/ssh2/crypto/cipher/BlockCipherFactory.java

```diff
@@ -1,115 +1,106 @@
  * @version $Id: BlockCipherFactory.java,v 1.2 2008/04/01 12:38:09 cplattne Exp $
  */
 public class BlockCipherFactory
 {
-	static class CipherEntry
-	{
-		String type;
-		int blocksize;
-		int keysize;
-		String cipherClass;
+    private static class CipherEntry
+    {
+        final String type;
+        final int blocksize;
+        final int keysize;
+        final String cipherClass;
 
-		public CipherEntry(String type, int blockSize, int keySize, String cipherClass)
-		{
-			this.type = type;
-			this.blocksize = blockSize;
-			this.keysize = keySize;
-			this.cipherClass = cipherClass;
-		}
-	}
+        CipherEntry(String type, int blockSize, int keySize, String cipherClass)
+        {
+            this.type = type;
+            this.blocksize = blockSize;
+            this.keysize = keySize;
+            this.cipherClass = cipherClass;
+        }
+    }
 
-	static Vector ciphers = new Vector();
+    private static final ArrayList<CipherEntry> ciphers = new ArrayList<>();
 
-	static
-	{
-		/* Higher Priority First */
+    static
+    {
+        /* Higher Priority First */
 
-		ciphers.addElement(new CipherEntry("aes256-ctr", 16, 32, "com.trilead.ssh2.crypto.cipher.AES"));
-		ciphers.addElement(new CipherEntry("aes192-ctr", 16, 24, "com.trilead.ssh2.crypto.cipher.AES"));
-		ciphers.addElement(new CipherEntry("aes128-ctr", 16, 16, "com.trilead.ssh2.crypto.cipher.AES"));
-		ciphers.addElement(new CipherEntry("blowfish-ctr", 8, 16, "com.trilead.ssh2.crypto.cipher.BlowFish"));
+        ciphers.add(new CipherEntry("aes256-ctr", 16, 32, "com.trilead.ssh2.crypto.cipher.AES$CTR"));
+        ciphers.add(new CipherEntry("aes192-ctr", 16, 24, "com.trilead.ssh2.crypto.cipher.AES$CTR"));
+        ciphers.add(new CipherEntry("aes128-ctr", 16, 16, "com.trilead.ssh2.crypto.cipher.AES$CTR"));
+        
+        ciphers.add(new CipherEntry("blowfish-ctr", 8, 16, "com.trilead.ssh2.crypto.cipher.BlowFish$CTR"));
 
-		ciphers.addElement(new CipherEntry("aes256-cbc", 16, 32, "com.trilead.ssh2.crypto.cipher.AES"));
-		ciphers.addElement(new CipherEntry("aes192-cbc", 16, 24, "com.trilead.ssh2.crypto.cipher.AES"));
-		ciphers.addElement(new CipherEntry("aes128-cbc", 16, 16, "com.trilead.ssh2.crypto.cipher.AES"));
-		ciphers.addElement(new CipherEntry("blowfish-cbc", 8, 16, "com.trilead.ssh2.crypto.cipher.BlowFish"));
-		
-		ciphers.addElement(new CipherEntry("3des-ctr", 8, 24, "com.trilead.ssh2.crypto.cipher.DESede"));
-		ciphers.addElement(new CipherEntry("3des-cbc", 8, 24, "com.trilead.ssh2.crypto.cipher.DESede"));
-	}
+        ciphers.add(new CipherEntry("aes256-cbc", 16, 32, "com.trilead.ssh2.crypto.cipher.AES$CBC"));
+        ciphers.add(new CipherEntry("aes192-cbc", 16, 24, "com.trilead.ssh2.crypto.cipher.AES$CBC"));
+        ciphers.add(new CipherEntry("aes128-cbc", 16, 16, "com.trilead.ssh2.crypto.cipher.AES$CBC"));
+        ciphers.add(new CipherEntry("blowfish-cbc", 8, 16, "com.trilead.ssh2.crypto.cipher.BlowFish$CBC"));
```

Comment:
I mean, if you're hardening this, then Blowfish is about as useful as 3DES. As in not at all.

Reply:
@jvz I am not an expert in these things. I mainly working with NETCONF protocol and for transport we use ssh (trilead ssh2)or tls in our client.  Server we are testing had some other kex alogorithms so my main focus was to just add support for them. What should be removed?